airflow/contrib/operators/dataflow_operator.py

```diff
@@ -128,6 +128,102 @@ def execute(self, context):
         hook.start_java_dataflow(self.task_id, dataflow_options, self.jar)
 
 
+class DataflowTemplateOperator(BaseOperator):
+    """
+    Start a Templated Cloud DataFlow batch job. The parameters of the operation
+    will be passed to the job.
+    It's a good practice to define dataflow_* parameters in the default_args of the dag
+    like the project, zone and staging location.
+    https://cloud.google.com/dataflow/docs/reference/rest/v1b3/LaunchTemplateParameters
+    https://cloud.google.com/dataflow/docs/reference/rest/v1b3/RuntimeEnvironment
+    ```
+    default_args = {
+        'dataflow_default_options': {
+            'project': 'my-gcp-project'
+            'zone': 'europe-west1-d',
+            'tempLocation': 'gs://my-staging-bucket/staging/'
+            }
+        }
+    }
+    ```
+    You need to pass the path to your dataflow template as a file reference with the
+    ``template`` parameter. Use ``parameters`` to pass on parameters to your job.
+    Use ``environment`` to pass on runtime environment variables to your job.
+    ```
+    t1 = DataflowTemplateOperator(
+        task_id='datapflow_example',
+        template='{{var.value.gcp_dataflow_base}}',
+        parameters={
+            'inputFile': "gs://bucket/input/my_input.txt",
+            'outputFile': "gs://bucket/output/my_output.txt"
+        },
+        dag=my-dag)
+    ```
+    ``template`` ``dataflow_default_options`` and ``parameters`` are templated so you can
+    use variables in them.
+    """
+    template_fields = ['parameters', 'dataflow_default_options', 'template']
+    ui_color = '#0273d4'
+
+    @apply_defaults
+    def __init__(
+            self,
+            template,
+            dataflow_default_options=None,
+            parameters=None,
```

Comment:
Fine with the current PR, but if you are open to more changes/nits,  maybe also remove parameters too and add back options for non-common inputs (e.g., options = { "parameters" : ...}).